site/documentation/content/admin-guide/secure_communication.md

```diff
@@ -127,6 +127,43 @@ In order to address this problem, the Netty networking library that is used in H
 
 The tcnative module comes in several flavors, corresponding to the way that the OpenSSL library has been linked in. The statically linked versions include a specific version of OpenSSL (or [BoringSSL](https://boringssl.googlesource.com/) for that matter) and is therefore most easy to use on supported platforms, regardless of whether another version of OpenSSL is already installed or not. In contrast, the dynamically linked variants depend on a particular version of OpenSSL being already installed on the operating system. Both approaches have their pros and cons and Hono therefore does not include tcnative in its Docker images by default, i.e. Hono's services will use the JVM's default SSL engine by default.
 
+## Server Name Indication (SNI)
+
+How Hono project is aimed to connect large number of devices in a secure way using primarily TLS. SNI, [TLS extension: server_name](https://tools.ietf.org/html/rfc6066#section-3), can be used to highly scale TLS connections.
+
+The most interesting use cases are Hono protocols adapters, please refer to their TLS specific configuration. 
+The use cases, when we need to serve multiple certificates can be differs, for example one of them can be : 
+
+* To ensure scalability on connecting large number of devices, the Hono platform can be accessible via different DNS domains or subdomains.
+* Public Key Infrastructure(PKI) for your IoT devices can have many trust stores.
+* Specific tenant TLS requirements.           
+
+Assuming that you have one of use cases that require SNI, the following steps are necessary to enable on Hono protocols adapters TLS SNI:
+
+* Certificate generation 
+
+Specific Hono protocol adapter should be configured to use key store, for example for MQTT adapter we have `--hono.mqtt.keyStorePath` for more info please refer to the specific protocol adapter configuration. 
+The key store can be only JKS or P12 format. Please note, that PEM format is not a key store and in this case PEM format will not work.
+The Common Name (CN), also known as the Fully Qualified Domain Name (FQDN), of yours certificate that are stored in key store should correspond to SNI name requested by the IoT device. 
```

Comment:
My understanding is that using the common name for the FQDN is discouraged these days in favor of adding a *subject alternative name* (SAN) property to the certificate. vert.x also considers this property when matching the SNI server name with a certificate. Based on that, I think we should rather recommend to use a corresponding SAN entry. WDYT?

Reply:
I also can see some ambiguity between what i wrote and vert.x documentation. I will update it.